tests/utils_tests/test_choices.py

```diff
@@ -0,0 +1,216 @@
+from unittest import mock
+
+from django.db.models import TextChoices
+from django.test import SimpleTestCase
+from django.utils.choices import CallableChoiceIterator, normalize_field_choices
+from django.utils.translation import gettext_lazy as _
+
+
+class NormalizeFieldChoicesTests(SimpleTestCase):
```

Comment:
I've added a fairly comprehensive set of tests. Hopefully it captures most cases 🤞🏻